pkg/autoscaler/autoscaler.go

```diff
@@ -457,6 +461,75 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 				cleanupObjects[asr.GetName()] = asr
 			}
 
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Waiting for MachineSet %s to acquire autoscaling annotations", machineSet.GetName()))
+				Eventually(komega.Object(machineSet), framework.WaitShort, pollingInterval).Should(HaveField("GetAnnotations()", SatisfyAll(
+					HaveKey(caMinSizeAnnotation),
+					HaveKey(caMaxSizeAnnotation),
+				)), "MachineSet %s did not acquire autoscaling annotation", machineSet.GetName())
+			}
+
+			// on some cloud providers, we have experienced the new nodes having resources added to their status.capacity
+			// after they have been initialized. this has been seen with `hugepages-1Gi` and `hugepages-2Mi`. so we
+			// wait until the nodes have both entries before moving on.
+			By("Waiting for the new Nodes to have hugepages-1Gi and hugepages-2Mi capacity")
+			Eventually(func() (bool, error) {
+				nodes := []*corev1.Node{}
+
+				for _, machineSet := range transientMachineSets {
+					if n, err := framework.GetNodesFromMachineSet(client, machineSet); err != nil {
+						return false, err
+					} else if len(n) != 1 {
+						return false, fmt.Errorf("expected 1 node for MachineSet %s, found %d", machineSet.GetName(), len(n))
+					} else {
+						nodes = append(nodes, n[0])
+					}
+				}
+
+				for _, n := range nodes {
+					_, ok1Gi := n.Status.Capacity[corev1.ResourceHugePagesPrefix+"1Gi"]
+					_, ok2Mi := n.Status.Capacity[corev1.ResourceHugePagesPrefix+"2Mi"]
+					if !ok1Gi || !ok2Mi {
+						return false, nil
+					}
+				}
+
+				return true, nil
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "Node capacity resources did not contain hugepages-1Gi and hugepages-2Mi")
```

Comment:
Potential rewrite to improve the output here, the improvement would be that if they don't match, the capacity map would be printed out so we could see what is there
```suggestion
			Eventually(func() ( ]*corev1.Node, error) {
				nodes := []*corev1.Node{}

				for _, machineSet := range transientMachineSets {
					if n, err := framework.GetNodesFromMachineSet(client, machineSet); err != nil {
						return false, err
					} else if len(n) != 1 {
						return false, fmt.Errorf("expected 1 node for MachineSet %s, found %d", machineSet.GetName(), len(n))
					} else {
						nodes = append(nodes, n[0])
					}
				}

				return nodes, nil
			}, framework.WaitMedium, pollingInterval).Should(HaveEach(
				HaveField("Status.Capacity", SatisfyAny(
					HaveKey(corev1.ResourceHugePagesPrefix+"1Gi"),
					HaveKey(corev1.ResourceHugePagesPrefix+"2Mi"),
				)),
			), "Node capacity resources did not contain hugepages-1Gi and hugepages-2Mi")
```

Reply:
i like printing the map when they don't match